hw_11/HW_1_Sotnikov.py

```diff
@@ -0,0 +1,368 @@
+
+    def __upper__(self):
+        return self.seq.upper()
+
+    def __lower__(self):
+        return self.seq.lower()
+
+    def correctness(self):
+        if any(monomer not in self.list for monomer in self.seq.upper()):
+            raise IncorrectSequenceAlphabet("Your sequence contains illegal symbols.")
+        else:
+            return True
+
+    def complement(self):
+        if self.correctness():
+            result = ''.join([self.dictionary[nucleotide] for nucleotide in self.seq])
+        return result
+
+    def gc_content(self):
+        gc_count = 0
+        for nucleotide in self.seq.upper():
+            gc_count += (nucleotide == "G" or nucleotide == "C")
+            gc_percent = (gc_count / len(self.seq)) * 100
+        return gc_percent
+
+## Creating an Aminoacid class
+
+class AminoAcidSequence(BiologicalSequence):
+    def __init__(self, seq):
+        self.seq = seq
+
+        self.list = ["A", "R", "N", "D", "C", "Q", "E", "G", "H",
+                    "I", "L", "K", "M", "F", "P", "S", "T", "W",
+                    "Y", "V", "O", "U", "B", "Z", "J"]
+
+        self.mass_dict = {
+            "A": 71, "R": 156, "N": 114, "D": 115, "C": 103, "Q": 128, "E": 129, "G": 57, "H": 137, "I": 113,
+            "L": 113, "K": 128, "M": 131, "F": 147, "P": 97, "S": 87, "T": 101, "W": 186, "Y": 163, "V": 99,
+            "O": 237, "U": 150, "B": 115, "Z": 129, "J": 115
+        }
+
+    def __len__(self):
+        return len(self.seq)
+
+    def __getitem__(self, slc):
+        return self.seq[slc]
+
+    def __str__(self):
+        return str(self.seq)
+
+    def __repr__(self):
+        return str(f"Sequence {self.seq}")
+
+    def __upper__(self):
+        return self.seq.upper()
+
+    def __lower__(self):
+        return self.seq.lower()
+
+    def correctness(self):
+        if any(monomer not in self.list for monomer in self.seq.upper()):
+            raise IncorrectSequenceAlphabet("Your sequence contains illegal symbols.")
+        else:
+            return True
+
+    def translate(self):
+        pass
+
+    def mol_mass(self):
+        result = sum([self.mass_dict[aminoacid] for aminoacid in self.seq.upper()])
+        return result
+
+## Creating two subclasses
+
+class DNASequence(NucleicAcidSequence):
+    def __init__(self, seq):
+        self.seq = seq
+
+        self.list = ["A", "G", "C", "T"]
+
+        self.dictionary = {
+            "a": "t", "A": "T", "t": "a", "T": "A", "g": "c", "G": "C", "c": "g", "C": "G",
+             "R": "Y", "r": "y", "Y": "R", "y": "r", "n": "n", "N": "N"
+            }
+
+        self.transcribed = {
+            "a": "a", "A": "A", "t": "u", "T": "U", "g": "g", "G": "G", "c": "c",
+            "C": "C", "R": "R", "r": "r", "Y": "Y", "y": "y", "n": "n", "N": "N"
+            }
+
+    def transcribe(self):
+        if self.correctness():
+            result = ''.join([self.transcribed[nucleotide] for nucleotide in self.seq])
+        return result
```

Comment:
Здесь же мы ожидаем РНК
```suggestion
            result = ''.join([self.transcribed[nucleotide] for nucleotide in self.seq])
        return RNASequence(result)
```